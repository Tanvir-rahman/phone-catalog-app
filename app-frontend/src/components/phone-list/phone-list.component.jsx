import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { phoneList } from '@store/phones/phone.selectors';

import PhoneListItem from '@components/phone-list-item/phone-list-item.component';
import './phone-list.styles.scss';

export const PhoneListComponent = ({ phones, history }) => {
  return (
    <div className="phone-list">
      {
        phones.map(({ id, ...otherPhoneProps }) => (
          <PhoneListItem 
            key={id}
            id={id}
            { ...otherPhoneProps }
          />
        ))
      }
    </div>
  );
}

const mapStateToProps = createStructuredSelector({
  phones: phoneList
});

export default withRouter(connect(mapStateToProps)(PhoneListComponent));