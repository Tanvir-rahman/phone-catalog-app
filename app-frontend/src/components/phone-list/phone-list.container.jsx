import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { isPhoneListFetching } from '@store/phones/phone.selectors';
import WithSpinner from '@components/with-spinner/with-spinner.component';
import PhoneListComponent from './phone-list.component';

const mapStateToProps = createStructuredSelector({
  isLoading: isPhoneListFetching
});

const PhoneListContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(PhoneListComponent);

export default PhoneListContainer;
