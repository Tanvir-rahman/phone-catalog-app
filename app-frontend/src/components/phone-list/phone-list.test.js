import React from 'react';
import { shallow } from 'enzyme';
import { PhoneListComponent } from './phone-list.component';

it('should render PhoneList component', () => {
  expect(shallow(<PhoneListComponent phones={[]} />)).toMatchSnapshot();
});
