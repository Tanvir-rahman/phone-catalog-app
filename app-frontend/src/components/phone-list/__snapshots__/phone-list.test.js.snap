// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PhoneList component 1`] = `
<div
  className="phone-list"
/>
`;
